tests/unit_tests/kd_ema_test.py

```diff
@@ -0,0 +1,142 @@
+import unittest
+from super_gradients.training.sg_model import SgModel
+from super_gradients.training.kd_model.kd_model import KDModel
+import torch
+from super_gradients.training.utils.utils import check_models_have_same_weights
+from super_gradients.training.datasets.dataset_interfaces.dataset_interface import ClassificationTestDatasetInterface
+from super_gradients.training.metrics import Accuracy
+from super_gradients.training.losses.kd_losses import KDLogitsLoss
+
+
+class KDModelTest(unittest.TestCase):
+    @classmethod
+    def setUp(cls):
+        cls.sg_trained_teacher = SgModel("sg_trained_teacher", device='cpu')
+        cls.dataset_params = {"batch_size": 5}
+        cls.dataset = ClassificationTestDatasetInterface(dataset_params=cls.dataset_params)
+
+        cls.kd_train_params = {"max_epochs": 3, "lr_updates": [1], "lr_decay_factor": 0.1, "lr_mode": "step",
+                               "lr_warmup_epochs": 0, "initial_lr": 0.1,
+                               "loss": KDLogitsLoss(torch.nn.CrossEntropyLoss()),
+                               "optimizer": "SGD",
+                               "criterion_params": {}, "optimizer_params": {"weight_decay": 1e-4, "momentum": 0.9},
+                               "train_metrics_list": [Accuracy()], "valid_metrics_list": [Accuracy()],
+                               "metric_to_watch": "Accuracy",
+                               'loss_logging_items_names': ["Loss", "Task Loss", "Distillation Loss"],
+                               "greater_metric_to_watch_is_better": True, "average_best_models": False,
+                               "ema": True}
+
+    def test_teacher_ema_not_duplicated(self):
+        """Check that the teacher EMA is a reference to the teacher net (not a copy)."""
+
+        kd_model = KDModel("test_teacher_ema_not_duplicated", device='cpu')
+        kd_model.connect_dataset_interface(self.dataset)
+        kd_model.build_model(student_architecture='resnet18',
+                             teacher_architecture='resnet50',
+                             student_arch_params={'num_classes': 1000},
+                             teacher_arch_params={'num_classes': 1000},
+                             checkpoint_params={'teacher_pretrained_weights': "imagenet"},
+                             run_teacher_on_eval=True, )
+
+        kd_model.train(self.kd_train_params)
+
+        assert kd_model.ema_model.ema.module.teacher is kd_model.net.module.teacher
+        assert kd_model.ema_model.ema.module.student is not kd_model.net.module.student
```

Comment:
use self.assert as done in other tests please.

Reply:
done in b161c58e646436229b624f587e3052e36382221e